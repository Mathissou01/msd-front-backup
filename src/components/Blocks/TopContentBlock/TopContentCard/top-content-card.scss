@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/02-tools/tools.include-media" as im;
@use "src/styles/02-tools/tools.shadows" as shadows;

.c-TopContentCard {
  $self: &;

  display: flex;
  flex-direction: column;
  width: 100%;
  margin-bottom: 9px;

  @include im.media(">=mobile") {
    margin-bottom: 12px;
  }

  @include im.media(">=tablet") {
    flex-direction: row;
    max-width: 768px;
    margin: 36px 60px 36px 20px;
  }

  @include im.media(">=laptop") {
    max-width: 784px;
    margin: 36px 0;
  }

  &__Banner {
    z-index: z("background");
    margin-right: -16px;
    margin-bottom: -26px;
    margin-left: -16px;
    overflow: hidden;

    @include im.media(">=515px") {
      margin-right: 0;
      margin-left: 0;
      border-radius: 8px;
    }

    @include im.media(">=mobile") {
      margin-bottom: -47px;
    }

    @include im.media(">=tablet") {
      min-width: 480px;
      height: 308px;
      margin: 0;
    }

    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
      transition: transform 0.2s ease-in-out;

      #{$self}:hover & {
        transform: scale(1.1);
      }
    }
  }

  &__CardContainer {
    z-index: z("site-content");
    overflow: hidden;
    background-color: colors.$white;
    border-radius: 8px;

    @include im.media(">=515px") {
      margin-right: 25px;
      margin-left: 25px;
    }

    @include im.media(">=tablet") {
      margin: 25px 0 25px -110px;

      #{$self}_styleEvent & {
        margin-left: -194px;
      }
    }
  }

  &__Card {
    @include shadows.shadow(1);

    position: relative;
    z-index: z("site-content");
    display: flex;
    flex-direction: column;
    gap: 16px;
    width: 100%;
    height: 100%;
    padding: 24px 16px;
    background-color: colors.$white;

    @include im.media(">=mobile") {
      padding: 24px 32px;
    }

    #{$self}:hover & {
      background-color: colors.$brand-primary-light;
    }
  }

  &__Header {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &__Tags {
    display: flex;
    gap: 6px;
    padding: 0;
  }

  &__Tag {
    @include typo.text("tag");

    padding: 6px 10px;
    color: colors.$expert-blue;
    border: 1px solid colors.$blue-4;
    border-radius: 4px;

    &_hasBackground {
      background-color: colors.$gray-0;
      border: hidden;
    }
  }

  &__Date {
    @include typo.text("tag");

    text-transform: lowercase;
  }

  &__Content {
    display: flex;
    flex-direction: column;
    gap: 6px;
    padding: 0;
  }

  &__Title {
    @include typo.heading(3);

    color: colors.$expert-blue;
  }

  &__Description {
    color: colors.$gray-800;
    font-weight: 400;
    font-size: 14px;
    line-height: 21px;
  }

  &__KnowMore {
    display: flex;
    gap: 12px;
    align-items: center;
    margin-top: auto;
    padding: 0;
    border-radius: 2px;
    transition: text-decoration 0.3s ease-in-out;

    #{$self}:hover & {
      text-decoration: underline;
    }

    span {
      @include typo.text("caption");

      color: colors.$blue-1;
    }
  }
}
