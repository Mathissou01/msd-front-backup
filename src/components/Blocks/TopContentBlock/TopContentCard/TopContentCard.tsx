import classNames from "classnames";
import React, { useEffect, useState } from "react";
import Link from "next/link";
import Image from "next/image";
import ArrowCorner from "public/images/pictos/arrow-corner.svg";
import {
  Maybe,
  TagEntity,
  UploadFileEntity,
} from "../../../../graphql/codegen/generated-types";
import {
  handleDateFrenchFormat,
  makePublicAssetPath,
} from "../../../../lib/utilities";
import "./top-content-card.scss";

interface ITopContentCardProps {
  title: string;
  shortDescription?: Maybe<string>;
  image: UploadFileEntity | null;
  href: string;
  tags?: TagEntity[];
  date?: string;
  style?: "styleEvent";
}

export default function TopContentCard({
  title,
  shortDescription,
  image,
  href,
  tags,
  date,
  style,
}: ITopContentCardProps) {
  const knowMore = "En savoir plus";
  const [formattedDate, setFormattedDate] = useState<string | null>(null);
  useEffect(() => {
    if (date) {
      setFormattedDate(handleDateFrenchFormat(new Date(date ?? "")));
    }
  }, [date]);

  const topContentCardClassNames = classNames("c-TopContentCard", {
    [`c-TopContentCard__Banner_${style}`]: style,
  });

  return (
    <Link className={topContentCardClassNames} href={href}>
      <div className="c-TopContentCard__Banner">
        {image?.attributes && (
          <Image
            src={makePublicAssetPath(image.attributes.url)}
            alt={image.attributes.alternativeText ?? ""}
            width={image.attributes.width ?? 375}
            height={image.attributes.height ?? 205}
          />
        )}
      </div>
      <div className="c-TopContentCard__CardContainer">
        <div className="c-TopContentCard__Card">
          <div className="c-TopContentCard__Header">
            <div className="c-TopContentCard__Tags">
              {tags?.map((tag, index) => (
                <span
                  key={index}
                  className="c-TopContentCard__Tag c-TopContentCard__Tag_hasBackground"
                >
                  {tag.attributes?.name}
                </span>
              ))}
            </div>
            {formattedDate && (
              <div className="c-TopContentCard__Date">{formattedDate}</div>
            )}
          </div>
          <div className="c-TopContentCard__Content">
            <h2 className="c-TopContentCard__Title">{title}</h2>
            <p className="c-TopContentCard__Description">{shortDescription}</p>
          </div>
          <div className="c-TopContentCard__KnowMore">
            <span>{knowMore}</span>
            <ArrowCorner />
          </div>
        </div>
      </div>
    </Link>
  );
}
