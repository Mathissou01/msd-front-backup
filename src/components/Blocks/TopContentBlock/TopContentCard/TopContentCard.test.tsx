import { render } from "@testing-library/react";
import TopContentCard from "./TopContentCard";

const mock = {
  title: "common top content card title",
  href: "/accueil",
  image: {
    attributes: {
      url: "/images/images-temp/temp_image.jpg",
      alternativeText: "",
      hash: "",
      mime: "",
      name: "",
      provider: "",
      size: 0,
    },
  },
};

describe("TopContentCard", () => {
  it("renders", () => {
    const { container } = render(
      <TopContentCard title={mock.title} href={mock.href} image={mock.image} />,
    );
    expect(container).toMatchSnapshot();
  });
});
