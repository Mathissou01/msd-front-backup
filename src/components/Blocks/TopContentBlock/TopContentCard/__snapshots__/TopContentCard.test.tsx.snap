// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopContentCard renders 1`] = `
<div>
  <a
    class="c-TopContentCard"
    href="/accueil"
  >
    <div
      class="c-TopContentCard__Banner"
    >
      <img
        alt=""
        data-nimg="1"
        decoding="async"
        height="205"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=750&q=75"
        srcset="/_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=750&q=75 2x"
        style="color: transparent;"
        width="375"
      />
    </div>
    <div
      class="c-TopContentCard__CardContainer"
    >
      <div
        class="c-TopContentCard__Card"
      >
        <div
          class="c-TopContentCard__Header"
        >
          <div
            class="c-TopContentCard__Tags"
          />
        </div>
        <div
          class="c-TopContentCard__Content"
        >
          <h3
            class="c-TopContentCard__Title"
          >
            common top content card title
          </h3>
          <p
            class="c-TopContentCard__Description"
          />
        </div>
        <div
          class="c-TopContentCard__KnowMore"
        >
          <span>
            En savoir plus
          </span>
          <div />
        </div>
      </div>
    </div>
  </a>
</div>
`;
